website/docs/configuration/types.html.md

```diff
@@ -0,0 +1,187 @@
+---
+layout: "docs"
+page_title: "Type Constraints - Configuration Language"
+sidebar_current: "docs-config-types"
+description: |-
+  Terraform module authors and provider developers can use detailed type
+  constraints to validate the inputs of their modules and resources.
+---
+
+# Type Constraints
+
+Terraform module authors and provider developers can use detailed type
+constraints to validate user-provided values for their input variables and
+resource arguments. This requires some additional knowledge about Terraform's
+type system, but allows you to build a more resilient user interface for your
+modules and resources.
```

Comment:
(Is this sufficient/accurate indication of whether you need to read this page or can get away with skipping it?)